### PR TITLE
fix: erase_after_first_eos 用 pad_token_id 填充，修复 GRPO loss 恒为 0

### PR DESCRIPTION
**Pull request by JingYao152**
erase_after_first_eos 原先把首次 EOS 之后的 token 替换成 eos_token_id， 而 create_attention_mask 仅按 != pad_token_id 判断有效位置，导致 EOS 填充位被误判为有效 token。这使得 batch 内所有 completion 的 mask 长度 都等于 batch 最长生成长度（min==max==mean），在 loss_type=bnpo + num_iterations=1（ratio=1）下 BNPO loss 化简为 -mean(adv)=0，loss 数值 恒为 0/-0。

改为用 pad_token_id 填充后，post-EOS 位置被正确识别为 padding，completion_mask.sum(1) 恢复为各样本真实长度，BNPO loss 在组内 reward 存在差异时不再恒为 0；组内 reward 全相同时 loss 仍为 0（正确的无信号行为）。同时避免了对 post-EOS 位置的错误 EOS 监督。

影响范围：仅 GRPO 训练（WeaverGRPOTrainer._generate_and_score_completions -> run_agent_loop）；SFT 走标准 SFTTrainer 不经过此函数，评测因skip_special_tokens=True 解码结果不变。

### CODE REVIEW

**Comment by JingYao152** [comment]
以下是运行 scripts/train/qwen2_5_gsm8k_grpo.sh 的部分训练日志：
{'loss': -0.2874, 'grad_norm': 6.230460166931152, 'learning_rate': 1.946508172362556e-06, 'num_tokens': 280752.0, 'completions/mean_length': 163.0, 'completions/min_length': 78.0, 'completions/max_length': 246.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 163.0, 'completions/min_terminated_length': 78.0, 'completions/max_terminated_length': 246.0, 'rewards/compute_reward/mean': 0.375, 'rewards/compute_reward/std': 0.5175492167472839, 'reward': 0.375, 'reward_std': 0.5175491571426392, 'frac_reward_zero_std': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 1.961367013372957e-06, 'num_tokens': 284827.0, 'completions/mean_length': 396.375, 'completions/min_length': 209.0, 'completions/max_length': 592.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 396.375, 'completions/min_terminated_length': 209.0, 'completions/max_terminated_length': 592.0, 'rewards/compute_reward/mean': 0.0, 'rewards/compute_reward/std': 0.0, 'reward': 0.0, 'reward_std': 0.0, 'frac_reward_zero_std': 1.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': 0.1183, 'grad_norm': 15.52566146850586, 'learning_rate': 1.976225854383358e-06, 'num_tokens': 287589.0, 'completions/mean_length': 220.25, 'completions/min_length': 59.0, 'completions/max_length': 371.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 220.25, 'completions/min_terminated_length': 59.0, 'completions/max_terminated_length': 371.0, 'rewards/compute_reward/mean': 0.25, 'rewards/compute_reward/std': 0.4629100561141968, 'reward': 0.25, 'reward_std': 0.4629100561141968, 'frac_reward_zero_std': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': -0.2042, 'grad_norm': 10.314806938171387, 'learning_rate': 1.9910846953937593e-06, 'num_tokens': 288971.0, 'completions/mean_length': 107.75, 'completions/min_length': 36.0, 'completions/max_length': 170.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 107.75, 'completions/min_terminated_length': 36.0, 'completions/max_terminated_length': 170.0, 'rewards/compute_reward/mean': 0.125, 'rewards/compute_reward/std': 0.3535533845424652, 'reward': 0.125, 'reward_std': 0.3535533845424652, 'frac_reward_zero_std': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 2.0059435364041606e-06, 'num_tokens': 291313.0, 'completions/mean_length': 198.75, 'completions/min_length': 10.0, 'completions/max_length': 508.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 198.75, 'completions/min_terminated_length': 10.0, 'completions/max_terminated_length': 508.0, 'rewards/compute_reward/mean': 0.0, 'rewards/compute_reward/std': 0.0, 'reward': 0.0, 'reward_std': 0.0, 'frac_reward_zero_std': 1.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': -0.0479, 'grad_norm': 7.848057270050049, 'learning_rate': 2.020802377414562e-06, 'num_tokens': 293578.0, 'completions/mean_length': 176.125, 'completions/min_length': 64.0, 'completions/max_length': 251.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 176.125, 'completions/min_terminated_length': 64.0, 'completions/max_terminated_length': 251.0, 'rewards/compute_reward/mean': 0.125, 'rewards/compute_reward/std': 0.3535533845424652, 'reward': 0.125, 'reward_std': 0.3535533845424652, 'frac_reward_zero_std': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': -0.2687, 'grad_norm': 1.9728059768676758, 'learning_rate': 2.035661218424963e-06, 'num_tokens': 295831.0, 'completions/mean_length': 179.625, 'completions/min_length': 31.0, 'completions/max_length': 363.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 179.625, 'completions/min_terminated_length': 31.0, 'completions/max_terminated_length': 363.0, 'rewards/compute_reward/mean': 0.25, 'rewards/compute_reward/std': 0.4629100561141968, 'reward': 0.25, 'reward_std': 0.4629100561141968, 'frac_reward_zero_std': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': 0.2375, 'grad_norm': 3.3630411624908447, 'learning_rate': 2.050520059435364e-06, 'num_tokens': 297497.0, 'completions/mean_length': 152.25, 'completions/min_length': 43.0, 'completions/max_length': 409.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 152.25, 'completions/min_terminated_length': 43.0, 'completions/max_terminated_length': 409.0, 'rewards/compute_reward/mean': 0.375, 'rewards/compute_reward/std': 0.5175492167472839, 'reward': 0.375, 'reward_std': 0.5175491571426392, 'frac_reward_zero_std': 0.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 2.0653789004457653e-06, 'num_tokens': 299966.0, 'completions/mean_length': 211.625, 'completions/min_length': 122.0, 'completions/max_length': 374.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 211.625, 'completions/min_terminated_length': 122.0, 'completions/max_terminated_length': 374.0, 'rewards/compute_reward/mean': 0.0, 'rewards/compute_reward/std': 0.0, 'reward': 0.0, 'reward_std': 0.0, 'frac_reward_zero_std': 1.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
{'loss': 0.0, 'grad_norm': 0.0, 'learning_rate': 2.0802377414561666e-06, 'num_tokens': 305339.0, 'completions/mean_length': 525.625, 'completions/min_length': 334.0, 'completions/max_length': 877.0, 'completions/clipped_ratio': 0.0, 'completions/mean_terminated_length': 525.625, 'completions/min_terminated_length': 334.0, 'completions/max_terminated_length': 877.0, 'rewards/compute_reward/mean': 0.0, 'rewards/compute_reward/std': 0.0, 'reward': 0.0, 'reward_std': 0.0, 'frac_reward_zero_std': 1.0, 'clip_ratio/low_mean': 0.0, 'clip_ratio/low_min': 0.0, 'clip_ratio/high_mean': 0.0, 'clip_ratio/high_max': 0.0, 'clip_ratio/region_mean': 0.0, 'epoch': 0.02}
